client/my-sites/site-settings/publishing-tools/post-by-email.tsx

```diff
@@ -0,0 +1,138 @@
+import { Button, FormLabel } from '@automattic/components';
+import { localizeUrl } from '@automattic/i18n-utils';
+import { ToggleControl } from '@wordpress/components';
+import { useTranslate } from 'i18n-calypso';
+import ClipboardButtonInput from 'calypso/components/clipboard-button-input';
+import FormFieldset from 'calypso/components/forms/form-fieldset';
+import SupportInfo from 'calypso/components/support-info';
+import JetpackModuleToggle from 'calypso/my-sites/site-settings/jetpack-module-toggle';
+import { useDispatch, useSelector } from 'calypso/state';
+import { activateModule, deactivateModule } from 'calypso/state/jetpack/modules/actions';
+import { regeneratePostByEmail } from 'calypso/state/jetpack/settings/actions';
+import isJetpackModuleActive from 'calypso/state/selectors/is-jetpack-module-active';
+import isJetpackModuleUnavailableInDevelopmentMode from 'calypso/state/selectors/is-jetpack-module-unavailable-in-development-mode';
+import isJetpackSiteInDevelopmentMode from 'calypso/state/selectors/is-jetpack-site-in-development-mode';
+import isRegeneratingJetpackPostByEmail from 'calypso/state/selectors/is-regenerating-jetpack-post-by-email';
+import isSiteAutomatedTransfer from 'calypso/state/selectors/is-site-automated-transfer';
+import { isJetpackSite } from 'calypso/state/sites/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
+type PostByEmailSettingProps = {
+	isFormPending: boolean;
+	address?: string;
+};
+
+const moduleSlug = 'post-by-email';
+
+export const PostByEmailSetting = ( { isFormPending, address }: PostByEmailSettingProps ) => {
+	const {
+		siteId: selectedSiteId,
+		siteIsJetpack: isJetpack,
+		siteIsAtomic: isAtomic,
+		regenerating: regeneratingPostByEmail,
+		isActive: active,
+		moduleIsUnavailable: moduleUnavailable,
+	} = useSelector( ( state ) => {
+		const siteId = getSelectedSiteId( state ) || 0;
+		const siteIsJetpack = isJetpackSite( state, siteId );
+		const siteIsAtomic = isSiteAutomatedTransfer( state, siteId );
+		const regenerating = isRegeneratingJetpackPostByEmail( state, siteId );
+		const isActive = !! isJetpackModuleActive( state, siteId, moduleSlug );
+		const siteInDevMode = isJetpackSiteInDevelopmentMode( state, siteId );
+		const moduleUnavailableInDevMode = isJetpackModuleUnavailableInDevelopmentMode(
+			state,
+			siteId,
+			moduleSlug
+		);
+
+		return {
+			siteId,
+			siteIsJetpack,
+			siteIsAtomic,
+			regenerating,
+			isActive,
+			moduleIsUnavailable: siteInDevMode && moduleUnavailableInDevMode,
+		};
+	} );
+
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const handleRegenerate = () => {
+		if ( ! selectedSiteId ) {
+			return;
+		}
+
+		const regenerate = regeneratePostByEmail( selectedSiteId );
+
+		dispatch( regenerate );
+	};
```

Comment:
Real talk: I'm very new to redux, and this approach came from reading and some trial and error. The fact that it works feels like cheating at worst and a little too easy at best. Please don't hesitate to to tell me where this has gone amusingly sideways. My brain is clay for you to mold 😆